src/component/Spinner.tsx

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { CircularProgress, makeStyles } from "@material-ui/core";
+
+const useStyles = makeStyles(_theme => ({
+  spinner: {
+    position: "absolute",
+    top: "50%",
+    left: "50%",
+    transform: "translate(-50%, -50%)"
+  }
+}));
+
+export default function Spinner() {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.spinner}>
+      <CircularProgress />
+    </div>
+  );
+}
```

Comment:
Can't you just center using flexbox? `style={{display: "flex", justifyContent: "center"}}` on the wrapper `div`. This adds a lot of JS processing which really isn't needed.

Reply:
I used flex box initially but noticed the spinner still being off-center when something else is rendered (The breadcrumb). This being absolutely positioned will make sure it's always perfectly centered. I could apply the styling in-line if that is preferred though.

Reply:
Ah, makes sense then. :+1: